src/app/desktop/components/Instances/index.js

```diff
@@ -23,9 +23,43 @@ const SubNoInstance = styled.div`
   font-size: 15px;
   margin-top: 20px;
 `;
+const sortAlphabetical = instances =>
+  instances.sort((a, b) => (a.name > b.name ? 1 : -1));
+
+const sortByLastPlayed = instances =>
+  instances.sort((a, b) => (a.lastPlayed < b.lastPlayed ? 1 : -1));
+
+const sortByMostPlayed = instances =>
+  instances.sort((a, b) => (a.timePlayed < b.timePlayed ? 1 : -1));
+
+const getInstances = (instances, sortOrder) => {
+  // Data normalization for missing fields
+  const inst = instances.map(instance => {
+    return {
+      ...instance,
+      timePlayed: instance.timePlayed || 0,
+      lastPlayed: instance.lastPlayed || 0
+    };
+  });
+
+  switch (sortOrder) {
+    case 0:
+      return sortAlphabetical(inst);
+    case 1:
+      return sortByLastPlayed(inst);
+    case 2:
+      return sortByMostPlayed(inst);
+    default:
+      return inst;
+  }
+};
 
 const Instances = () => {
-  const instances = useSelector(_getInstances);
+  // const instances = getInstances();
+  const instanceSortOrder = useSelector(
+    state => state.settings.instanceSortOrder
+  );
+  const instances = getInstances(useSelector(_getInstances), instanceSortOrder);
```

Comment:
In general, it looks good. The only thing I would change is this part here.

What I would do is have `instanceSortOrder` and `instances` come directly from the state.
Then, below, have a `memoInstances` which has a `useMemo` with both `instances` and `instanceSortOrder` as deps and it returns the `getInstances(instances || [], instanceSortOrder)`

Reply:
Makes perfect sense - I have implemented this change in 43b8017437dd023ab2b09417e60fd0cc7777a858